File: memory/memory.go
package main

import (
	"fmt"
	"runtime"
	time "time"
)

type Payload struct {
	data [1024]byte
}

var Sink *Payload

func main() {
	count := 10_000_000 
	start := time.Now()
	
	for i := 0; i < count; i++ {
		p := &Payload{}
		Sink = p 
	}

	var m runtime.MemStats
	runtime.ReadMemStats(&m)
	
	fmt.Printf("Go: Allocations took %v\n", time.Since(start))
	fmt.Printf("Go: NumGC = %v\n", m.NumGC)
}

File: memory/memory.rs
use std::time::Instant;

#[allow(dead_code)] 
struct Payload {
    data: [u8; 1024],
}

impl Payload {
    fn new() -> Self {
        Self { data: [0; 1024] }
    }
}

fn main() {
    let count = 10_000_000;
    let start = Instant::now();

    for _ in 0..count {
        let _p = Box::new(Payload::new()); 
        // Scope end: Immediate drop (free)
    }

    println!("Rust: Allocations took {:?}", start.elapsed());
}
